src/components/formStore/table/useFormStoreTable.tsx

```diff
@@ -391,11 +391,27 @@ export default function useFormStoreTable({
     submissionIdValidationMessage,
   ])
 
+  const defaultHiddenColumns = [
+    'SUBMISSION_ID',
+    'EXTERNAL_ID',
+    'TASK',
+    'TASK_ACTION',
+    'TASK_GROUP',
+    'TASK_GROUP_INSTANCE',
+  ]
+
   const [initialState] = React.useState<Partial<TableState<FormStoreRecord>>>(
     () => {
       const text = localStorage.getItem(localStorageKey(form.id))
       if (text) {
-        return JSON.parse(text)
+        const intialState = JSON.parse(text)
+
+        if (intialState && !intialState.defaultHiddenColumnsUpToDate) {
+          intialState.hiddenColumns = defaultHiddenColumns
```

Comment:
We need to concat the defaults with the current `hiddenColumns` otherwise all the form element columns they have hidden will reappear.

Reply:
mmm, will also need to ensure that if the current user has re-enabled the `SUBMISSION_ID` and/or `EXTERNAL_ID`, they we are not hiding them.